crates/bitwarden-core/src/client/internal.rs

```diff
@@ -315,6 +318,31 @@ impl InternalClient {
         self.initialize_user_crypto_decrypted_key(decrypted_user_key, key_state)
     }
 
+    #[cfg(feature = "internal")]
+    pub(crate) fn initialize_user_crypto_pin_envelope(
+        &self,
+        pin: String,
+        pin_protected_user_key_envelope: PasswordProtectedKeyEnvelope,
+        key_state: UserKeyState,
+    ) -> Result<(), EncryptionSettingsError> {
+        let decrypted_user_key = {
+            // Note: This block ensures ctx is dropped. Otherwise it would cause a deadlock when
+            // initializing the user crypto
+            use crate::key_management::SymmetricKeyId;
+            let ctx = &mut self.key_store.context_mut();
+            let decrypted_user_key_id = pin_protected_user_key_envelope
+                .unseal(SymmetricKeyId::Local("tmp_unlock_pin"), &pin, ctx)
```

Comment:
@dani-garcia did we have any convention for local IDs? Or is it just the wild west.

Reply:
I think the approach here is to push dani's other work (https://bitwarden.atlassian.net/browse/PM-18102), so that we don't have to deal with naming collisions..

Reply:
Yeah, I'd like to see that merged so we don't have to care about naming in situations like this.